cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/ValidatorImpl.java

```diff
@@ -0,0 +1,207 @@
+        List<RuntimeException> errors = new ArrayList<>();
+        if(executable.getType().equals(SlangTextualKeys.OPERATION_TYPE)){
+            return errors;
+        }
+        Flow flow = (Flow)executable;
+        Deque<Step> steps = flow.getWorkflow().getSteps();
+        for(Step step : steps){
+            List<Map<String, String>> stepNavigations = step.getNavigationStrings();
+            String refId = step.getRefId();
+            Executable reference = filteredDependencies.get(refId);
+            if (reference == null) {
+                errors.add(new IllegalArgumentException("Cannot compile flow: \'" + executable.getName() + "\' since for step: \'" + step.getName()
+                        + "\', the dependency: \'" + refId + "\' is missing."));
+            } else {
+                List<Result> refResults = reference.getResults();
+                for(Result result : refResults){
+                    String resultName = result.getName();
+                    if (!navigationListContainsKey(stepNavigations, resultName)){
+                        errors.add(new IllegalArgumentException("Cannot compile flow: \'" + executable.getName() +
+                                "\' since for step: '" + step.getName() + "\', the result \'" + resultName+
+                                "\' of its dependency: \'"+ refId + "\' has no matching navigation"));
+                    }
+                }
+            }
+        }
+        return errors;
+    }
+
+    private boolean navigationListContainsKey(List<Map<String, String>> stepNavigations, String resultName) {
+        for (Map<String, String> map : stepNavigations) {
+            if (map.containsKey(resultName)) return true;
+        }
+        return false;
+    }
+
+    private List<RuntimeException> validateRequiredNonPrivateInputs(
+            Executable executable,
+            Map<String, Executable> dependencies,
+            Set<Executable> verifiedExecutables,
+            List<RuntimeException> errors) {
+        //validate that all required & non private parameters with no default value of a reference are provided
+        if(!SlangTextualKeys.FLOW_TYPE.equals(executable.getType()) || verifiedExecutables.contains(executable)){
+            return errors;
+        }
+        verifiedExecutables.add(executable);
+
+        Flow flow = (Flow) executable;
+        Collection<Step> steps = flow.getWorkflow().getSteps();
+        Set<Executable> flowReferences = new HashSet<>();
+
+        for (Step step : steps) {
+            Executable reference = dependencies.get(step.getRefId());
+            List<String> mandatoryInputNames = getMandatoryInputNames(reference);
+            List<String> stepInputNames = getStepInputNames(step);
+            List<String> inputsNotWired = getInputsNotWired(mandatoryInputNames, stepInputNames);
+
+            try {
+                validateInputNamesEmpty(inputsNotWired, flow, step, reference);
```

Comment:
`validateRequiredNonPrivateInputs` name is not correct

This method could get a new role, namely going over the flow recursively and do three validations: matching navigations (+ recursive test), inputs are supplied, inputs are different

WDYT?